doc/reference/misc/notify.rst

```diff
@@ -0,0 +1,25 @@
+.. _async_notification:
```

Comment:
this could be called `sys_notify` now to be aligned with `sys_notify_apis`

Reply:
It can't: the target name `sys_notify` refers to `struct sys_notify`.